src/psyclone/tests/psyGen_test.py

```diff
@@ -3594,6 +3943,59 @@ def test_kernelschedule_name_setter():
     assert kschedule.name == "newname"
 
 
+def test_kernelschedule_create():
+    '''Test that the create method in the KernelSchedule class correctly
+    creates a KernelSchedule instance.
+
+    '''
+    symbol_table = SymbolTable()
+    symbol_table.add(DataSymbol("tmp", "real"))
+    kschedule = KernelSchedule.create("mod_name", symbol_table,
+                                      [Assignment.create(Reference("tmp"),
+                                                         Literal("0.0"))])
+    result = FortranWriter().kernelschedule_node(kschedule)
+    assert result == (
+        "subroutine mod_name()\n"
+        "  real :: tmp\n\n"
+        "  tmp=0.0\n\n"
+        "end subroutine mod_name\n")
+
+
+def test_kernelschedule_create_invalid():
+    '''Test that the create method in a KernelSchedule class raises the
+    expected exception if the provided input is invalid.
+
+    '''
+    symbol_table = SymbolTable()
+    symbol_table.add(DataSymbol("x", "real"))
+    children = [Assignment.create(Reference("x"), Literal("1"))]
+
+    # name is not a string.
+    with pytest.raises(GenerationError) as excinfo:
+        _ = KernelSchedule.create(1, symbol_table, children)
+    assert ("name argument to class KernelSchedule method create "
+            "should be a string but found 'int'.") in str(excinfo.value)
+
+    # symbol_table not a SymbolTable.
+    with pytest.raises(GenerationError) as excinfo:
+        _ = KernelSchedule.create("mod_name", "invalid", children)
+    assert ("symbol_table argument to class KernelSchedule method create "
+            "should be a SymbolTable but found 'str'.") in str(excinfo.value)
+
+    # children not a list.
+    with pytest.raises(GenerationError) as excinfo:
+        _ = KernelSchedule.create("mod_name", symbol_table, "invalid")
+    assert ("children argument to class KernelSchedule method create should "
+            "be a list but found 'str'." in str(excinfo.value))
+
+    # contents of children list are not Node.
+    with pytest.raises(GenerationError) as excinfo:
+        _ = KernelSchedule.create("mod_name", symbol_table, ["invalid"])
+    assert (
+        "child of children argument to class KernelSchedule method create "
+        "should be a PSyIR Node but found 'str'." in str(excinfo.value))
+
+
```

Comment:
Good job writing tests for all the possible input errors, as a general comment to all the error messages I am not sure if "children argument to class X method create should be ..." sounds good. Shouldn't it be "children argument of create method in/from class X should be ..."

Reply:
Well that involved a lot of search and replace.